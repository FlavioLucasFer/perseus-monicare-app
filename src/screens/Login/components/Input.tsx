import { ErrorMessage, useField } from 'formik';
import React, { useState } from 'react';
import { 
	StyleSheet, 
	Text, 
	TextInput, 
	TextInputProps, 
	View, 
} from "react-native";
import { Icon } from 'react-native-elements';

interface LoginInputProps extends TextInputProps {
	name: string,
	placeholder: string,
};

const styles = StyleSheet.create({
	container: {
		display: 'flex',
		flexDirection: 'row',
		alignItems: 'center',

		backgroundColor: '#F6F6F6',
		borderWidth: 1, 
		borderColor: '#BDBDBD', 
		borderRadius: 7,
	},
	input: {
		paddingLeft: 15,
		fontSize: 16,
		width: '90%',
	},
	errorMessage: {
		color: '#F00',
	},
});

const Input: React.FC<LoginInputProps> = props => {
	const [field, meta, helpers] = useField(props.name);
	const [isPasswordVisible, setIsPasswordVisible] = useState(false);
	const [secureTextEntry, setSecureTextEntry] = useState(props.secureTextEntry);

	function handleVisibilityIconPress() {
		setIsPasswordVisible(!isPasswordVisible);
		setSecureTextEntry(!secureTextEntry);
	}

	return (
		<>
			<View 
				style={styles.container}
			>
				<TextInput 
					{...props}
					value={field.value}
					placeholderTextColor='#BDBDBD'
					style={[styles.input, props.style]}
					onBlur={() => helpers.setTouched(!meta.touched)}
					onChangeText={helpers.setValue}
					secureTextEntry={secureTextEntry}
				/>
				{props.secureTextEntry &&
					<Icon
						name={isPasswordVisible ? 'visibility-off' : 'visibility'}
						color={isPasswordVisible ? '#0000004D' : undefined}
						type='material'
						size={28}
						onPress={handleVisibilityIconPress}
					/>
				}
			</View>
			<Text style={styles.errorMessage}>
				<ErrorMessage name={props.name} />
			</Text>
		</>
	);
}; 

export type {
	LoginInputProps,
};
export default Input;
