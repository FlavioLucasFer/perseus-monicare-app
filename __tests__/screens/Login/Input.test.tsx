import React from 'react';
import { shallow } from 'enzyme';

import Input from 'screens/Login/components/Input';

let mockField = {};
let mockMeta = {};
let mockHelpers = {};

jest.mock("formik", () => ({
	...jest.requireActual("formik"),
	useField: jest.fn(() => {
		return [mockField, mockMeta, mockHelpers];
	}),
}));

describe('Input component', () => {
	it('Should render properly', () => {
		mockField = { name: "test" };
		mockMeta = { touched: false, error: undefined };
		mockHelpers = { setValue: jest.fn() };
		const defaultProps = { name: "test", onChange: jest.fn(), status: "default" };

		const wrapper = shallow(
			<Input 
				{...defaultProps}
				placeholder='Test'
			/>
		);
		expect(wrapper.debug()).toMatchSnapshot();
	});
});
