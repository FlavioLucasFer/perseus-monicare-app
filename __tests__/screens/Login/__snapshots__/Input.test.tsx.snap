// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Should render properly 1`] = `
"<Fragment>
  <View style={{...}}>
    <TextInput name=\\"test\\" onChange={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} status=\\"default\\" placeholder=\\"Test\\" value={[undefined]} placeholderTextColor=\\"#BDBDBD\\" style={{...}} onBlur={[Function: onBlur]} onChangeText={[Function: mockConstructor] { _isMockFunction: true, getMockImplementation: [Function (anonymous)], mock: Object [Object: null prototype] { calls: [], instances: [], invocationCallOrder: [], results: [] }, mockClear: [Function (anonymous)], mockReset: [Function (anonymous)], mockRestore: [Function (anonymous)], mockReturnValueOnce: [Function (anonymous)], mockResolvedValueOnce: [Function (anonymous)], mockRejectedValueOnce: [Function (anonymous)], mockReturnValue: [Function (anonymous)], mockResolvedValue: [Function (anonymous)], mockRejectedValue: [Function (anonymous)], mockImplementationOnce: [Function (anonymous)], mockImplementation: [Function (anonymous)], mockReturnThis: [Function (anonymous)], mockName: [Function (anonymous)], getMockName: [Function (anonymous)] }} secureTextEntry={[undefined]} />
  </View>
  <Text style={{...}}>
    <FormikConnect(ErrorMessageImpl) name=\\"test\\" />
  </Text>
</Fragment>"
`;
